File: auth/dto/UserDTO.go
package dto

import (
	"auth_service/model"

	"github.com/google/uuid"
)

type UserDTO struct {
	ID uuid.UUID `json:"id"`
	Username string `json:"username"`
	Email string `json:"email"`
	UserRole string `json:"role"`
	ImagePath string `json:"image_path"`
	FirstName string `json:"first_name"`
	LastName string `json:"last_name"`
	Bio string `json:"bio"`
	Motto string `json:"motto"`
}

func UserToDTO(user *model.User) UserDTO {
	var dto UserDTO
	dto.ID = user.ID
	dto.Username = user.Username
	dto.Email = user.Email
	dto.UserRole = user.UserRole.String()
	dto.ImagePath = user.ImagePath 
	dto.FirstName = user.FirstName 
	dto.LastName = user.LastName 
	dto.Bio = user.Bio 
	dto.Motto = user.Motto 
	return dto
}

File: follower-service/src/main/java/com/example/follower_service/dto/UserDTO.java
package com.example.follower_service.dto;

import com.example.follower_service.model.User;

public class UserDTO {
    private String id;
    private String username;

    public UserDTO() {}

    public UserDTO(User u) {
        id = u.getId();
        username = u.getUsername();
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }
}
